docs/style.rst

```diff
@@ -11,6 +11,25 @@ and follow it when working on Nengo projects.
 Python
 ======
 
+We use ``black`` for automated formatting.
+We recommend that you set up your editor to run ``black``
+when you save a Python file;
+see `Black editor integration
+<https://github.com/psf/black#editor-integration>`__
+for details on how to do that.
+
+Whether you set up ``black`` with your editor or not,
+we require that you run ``black`` through a pre-commit hook.
```

Comment:
Do we need to require this? It seems like if someone prefers to do it through editor configurations (or even to run black manually), it doesn't make much difference to us. Since we're double checking that everything was run correctly in the static checks, in the worst case the only pain will fall on the contributor (who will have to re-run black and re-commit).

Reply:
We don't need to, but I do think it's good practice because the pre-commit hook is very fast while pushing and waiting for a TravisCI build is not very fast. The pre-commit hook only runs the check on the files that are modified rather than the whole repository. I think if people get used to running pre-commit hooks (which are super simple once you do it once) then we can add more stuff in there and save ourselves time long-term.

That's my opinion anyhow, this is something I'd probably let the consensus decide.

Reply:
Not to jump in on this, but I think that the pre-commit should be a required step. If they don't want to set it up in their editor, then I think that should be up to them, although I don't see why they wouldn't want to.

Reply:
I agree with Dan that I don't think we need to require this, though we should encourage it perhaps. I see the `black` check just like our other static checks; it certainly helps to somehow do them yourself before you commit, but it's not a requirement. I guess the downside to this is that somebody could have a commit that fails `black` that is fixed by a later commit, and ideally we'd have every commit passing `black` (and all the other static checks for that matter).

When it comes down to it, though, I'm not sure if it really matters what language we use here. There's no way to enforce that somebody does it, and if we're strongly encouraging people to use it, I'm fine to just say it's "required". Within our team I think we should require it. People who are infrequent contributors to Nengo might not even be reading this document, and I don't think using the word "required" is going to deter anybody from contributing to Nengo because of the hassle of setting up pre-commit hooks.

Reply:
Pushed a fixup to slightly change the wording. Still indicating that it is an expected part of our workflow, just not quite as "aggressive" as making it a requirement.  Feel free to revert if you want to stick with the "required" language, I'm fine either way!